docs/ocis/adr/0011-global-url-format.md

```diff
@@ -0,0 +1,210 @@
+| `https://demo.owncloud.com/#/s/b78c2044-5b51-446f-82f6-907a664d089c:194b4a97-597c-4461-ab56-afd4f5a21607/relative/path/to/resource` | sub folder `/relative/path/to/resource` in the storage with id `b78c2044-5b51-446f-82f6-907a664d089c:194b4a97-597c-4461-ab56-afd4f5a21607`, works ***only*** if path still exists  |
+| `https://demo.owncloud.com/#/s/b78c2044-5b51-446f-82f6-907a664d089c:194b4a97-597c-4461-ab56-afd4f5a21607/relative/path/to/resource?id=ba4c1820-df12-11eb-8dcd-ff21f12c1264:beb78dd6-df12-11eb-a05c-a395505126f6` | sub folder `/relative/path/to/resource` in the storage with id `b78c2044-5b51-446f-82f6-907a664d089c:194b4a97-597c-4461-ab56-afd4f5a21607`, lookup can fall back to the `id` |
+
+{{< hint >}}
+* `/#` is used by the current vue router.
+* `/s` denotes that this is a space url.
+* `<space_id>` and `<resource_id>` both consist of `<storage_id>:<node_id>`, but the `space_id` can be replaced with a shorter id or an alias. See furthor down below.
+* `<relative/path>` takes precedence over the `<resource_id>`, both are optional
+{{< /hint >}}
+
+* Good, because the URLs reveal a relevant path context to users
+* Good, because everything after the `#` is not sent to the server, building the webdav request to list the folder is offloaded to the clients
+* Good, because powerusers can navigate by updating the path in the url
+* Bad, because the current ids are uuid based, leading to very long URLs where tha path component nearly vanishes between two very long strings
+* Bad, because the `#` in the URL is just a technical requirement
+* Bad, because ocis web requires a `/#/files/s` at the root of the route to distinguish the files app from other apps
+* Bad, while navigating using the WebUI, the URL has to be updated whenever we change spaces.
+
+With the above explained, let's see some use cases:
+
+#### Example 1: UserA shares something from her Home folder with UserB
+
+- open the browser and go to `demo.owncloud.com`
+- the browser's url changes to: `https://demo.owncloud.com/#/s/b78c2044-5b51-446f-82f6-907a664d089c:194b4a97-597c-4461-ab56-afd4f5a21607`. You're now in YOUR home folder / personal space.
+- you create a new folder `/relative/path/to/resource` and navigate into `/relative/path/to`
+  - the URL now changes to: `https://demo.owncloud.com/#/s/b78c2044-5b51-446f-82f6-907a664d089c:194b4a97-597c-4461-ab56-afd4f5a21607/relative/path/to`
+- You share `resource` with some else
+- YOU navigate into `/relative/path/to/resource`
+  - now the URL would look like: `https://demo.owncloud.com/#/s/b78c2044-5b51-446f-82f6-907a664d089c:3a9305da-df17-11eb-ab99-abe09d93e08a`
+
+As you can see, even if you're the owner of `/relative/path/to/resource` and navigate into it, the URL changes due to a new space being entered. This ensures that while working in your home folder, copying URLs and giving them to the person you share the resource with, the receiver can still navigate within the new space.
```

Comment:
Well written; it coincides with my previous comment 👍

Is it worth mentioning that this is at least how it works in the ocis storage? I don't know how this could work without ACLs. I also haven't spent the time doing the mental effort thinking about it hard enough 🤯